From c325bc355533db9e7eea7ca746ce3f668ef480ee Mon Sep 17 00:00:00 2001
From: Daniel Mach <dmach@redhat.com>
Date: Tue, 7 May 2019 13:25:18 +0200
Subject: [PATCH] Fix 'no such table: main.trans_cmdline' error (RhBug:1596540)

The crash was caused by opening .sqlite-journal file
instead of .sqlite database. Skipping files with
invalid suffix fixes the problem.
---
 libdnf/transaction/Transformer.cpp | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/libdnf/transaction/Transformer.cpp b/libdnf/transaction/Transformer.cpp
index 026d79f..0cb12c7 100644
--- a/libdnf/transaction/Transformer.cpp
+++ b/libdnf/transaction/Transformer.cpp
@@ -35,6 +35,7 @@
 
 #include "../utils/bgettext/bgettext-lib.h"
 #include "../utils/filesystem.hpp"
+#include "../utils/utils.hpp"
 
 #include "RPMItem.hpp"
 #include "Swdb.hpp"
@@ -658,12 +659,11 @@ Transformer::historyPath()
         throw Exception(_("Transformer: can't open history persist dir"));
     }
 
-    // iterate over history directory
+    // iterate over history directory and look for 'history-*.sqlite' files
     while ((dp = readdir(dirp.get())) != nullptr) {
         std::string fileName(dp->d_name);
-
-        // push the possible history DB files into vector
-        if (fileName.find("history") != std::string::npos) {
+        if (libdnf::string::startsWith(fileName, "history-") &&
+            libdnf::string::endsWith(fileName, ".sqlite")) {
             possibleFiles.push_back(fileName);
         }
     }
--
libgit2 0.28.2

